engine/runtime-instrument-id-execution/src/main/java/org/enso/interpreter/instrument/IdExecutionInstrument.java

```diff
@@ -246,11 +247,13 @@ public void onReturnExceptional(VirtualFrame frame, Throwable exception) {
         onReturnValue(frame, new PanicSentinel(panicException, context.getInstrumentedNode()));
       } else if (exception instanceof PanicSentinel) {
         onReturnValue(frame, exception);
+      } else if (exception instanceof AbstractTruffleException) {
+        onReturnValue(frame, exception);
       }
     }
 
     private void onExpressionReturn(Object result, Node node, EventContext context) throws ThreadDeath {
-      boolean isPanic = result instanceof PanicSentinel;
+      boolean isPanic = result instanceof AbstractTruffleException;
```

Comment:
do we not need to check for `PanicSentinel` anymore?
Could we remove lines 248-249 as well then?

Reply:
`PanicSentinel extends AbstractTruffleException` - so this check covers also `PanicSentinel`.